File: python/leetcode/0003-longest-substring-without-repeating-characters/solution01.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxlen = 0
        sequence = ""

        for c in s:
            if c in sequence:
                maxlen = max(maxlen, len(sequence))
                i = sequence.index(c)
                sequence = sequence[i + 1::] + c
            else:
                sequence += c

        return max(maxlen, len(sequence))

File: typescript/leetcode/0003-longest-substring-without-repeating-characters/solution01.ts
/**
 * @param {string} s
 * @return {number}
 */
export function lengthOfLongestSubstring(s: string): number {
  let maxlen = 0
  let sequence = ""

  for (let i = 0; i < s.length; i++) {
    const c = s[i]
    if (sequence.indexOf(c) !== -1) {
      maxlen = Math.max(maxlen, sequence.length)

      const index = sequence.indexOf(c)
      sequence = sequence.substring(index + 1) + c
    } else {
      sequence += c
    }
  }

  return Math.max(maxlen, sequence.length)
}
